app/src/main/java/com/globant/weatherapp/MainActivity.kt

```diff
@@ -2,11 +2,27 @@ package com.globant.weatherapp
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.widget.ArrayAdapter
+import android.widget.Spinner
+import kotlinx.android.synthetic.main.city_layout.city_selection_spinner
+import kotlinx.android.synthetic.main.city_layout.country_selection_spinner
 
 class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContentView(R.layout.city_layout)
+
+        setAdapterToSpinner(R.array.country_array, R.layout.spinner_item, R.layout.spinner_dropdown_item, country_selection_spinner)
```

Comment:
where is the MVP structure?? you have to call the presenter.init() for example and is the presenter that have to cal view.initAdapter for each spinner

Reply:
You are right, thanks Mau!